src/sql/workbench/contrib/assessment/common/asmtActions.ts

```diff
@@ -0,0 +1,227 @@
+		this._logService.info(`Requesting server items`);
+		return this._assessmentService.assessmentInvoke(ownerUri, AssessmentTargetType.Server);
+	}
+
+	getDatabaseItems(ownerUri: string): Thenable<AssessmentResult> {
+		return this._assessmentService.assessmentInvoke(ownerUri, AssessmentTargetType.Database);
+	}
+}
+
+export class AsmtDatabaseInvokeItemsAction extends Action {
+	public static ID = 'asmtaction.database.invokeitems';
+
+	constructor(
+		databaseName: string,
+		@IAssessmentService private _assessmentService: IAssessmentService,
+		@IAdsTelemetryService private _telemetryService: IAdsTelemetryService
+	) {
+		super(AsmtDatabaseInvokeItemsAction.ID,
+			nls.localize('asmtaction.database.invokeitems', "Invoke Assessment for {0}", databaseName),
+			TARGET_ICON_CLASS[AssessmentTargetType.Database]);
+	}
+
+	public async run(context: IAsmtActionInfo): Promise<boolean> {
+		this._telemetryService.sendActionEvent(SqlAssessmentTelemetryView, this.id);
+		if (context && context.component) {
+			context.component.showProgress(AssessmentType.InvokeAssessment);
+			let dbAsmtResults = await this._assessmentService.assessmentInvoke(context.ownerUri, AssessmentTargetType.Database);
+			context.component.showInitialResults(dbAsmtResults, AssessmentType.InvokeAssessment);
+			context.component.stopProgress(AssessmentType.InvokeAssessment);
+			return Promise.resolve(true);
+		}
+		return Promise.resolve(false);
+	}
+}
+
+export class AsmtExportAsScriptAction extends Action {
+	public static ID = 'asmtaction.exportasscript';
+	public static LABEL = nls.localize('asmtaction.exportasscript', "Export As Script");
+
+	constructor(
+		@IAssessmentService private _assessmentService: IAssessmentService,
+		@IAdsTelemetryService private _telemetryService: IAdsTelemetryService
+	) {
+		super(AsmtExportAsScriptAction.ID, AsmtExportAsScriptAction.LABEL, 'exportAsScriptIcon');
+	}
+
+	public async run(context: IAsmtActionInfo): Promise<boolean> {
+		this._telemetryService.sendActionEvent(SqlAssessmentTelemetryView, AsmtExportAsScriptAction.ID);
+		if (context && context.component && context.component.resultItems) {
+			await this._assessmentService.generateAssessmentScript(context.ownerUri, context.component.resultItems);
+			return Promise.resolve(true);
+		}
+		return Promise.resolve(false);
+	}
+}
+
+export class AsmtSamplesLinkAction extends Action {
+	public static readonly ID = 'asmtaction.showsamples';
+	public static readonly LABEL = nls.localize('asmtaction.showsamples', "View all rules and learn more on GitHub");
+	public static readonly ICON = 'asmt-learnmore';
+	private static readonly configHelpUri = 'https://aka.ms/sql-assessment-api';
+
+	constructor(
+		@IOpenerService private _openerService: IOpenerService,
+		@IAdsTelemetryService private _telemetryService: IAdsTelemetryService
+
+	) {
+		super(AsmtSamplesLinkAction.ID, AsmtSamplesLinkAction.LABEL, AsmtSamplesLinkAction.ICON);
+	}
+
+	public run(): Promise<boolean> {
```

Comment:
Mark this as async.

Rationale: https://palantir.github.io/tslint/rules/promise-function-async/